### ISSUE TITLE
Returning new lines in document symbol names is breaking some tooling

### ISSUE THREAD

**Issue by idursun** [lexical-lsp/lexical#808]
I changed my NeoVim configuration to use`lexical` instead of `elixir-ls` today and noticed an error message in Telescope picker when displaying document symbols.

I discovered that error only occurs when the module contains function definitions spanning multiple lines.

For example displaying document symbols in the following module causes an error:

```elixir
defmodule Module do
  def my_fn(
     hello
  ) do
  end
end
```

the symbol is parsed like the following

```
%Lexical.RemoteControl.CodeIntelligence.Symbols.Document{
  name: "def my_fn(\n     hello\n  )",
  type: {:function, :public},
  range: LxRange[<<2, 3>>...<<5, 6>>],
  detail_range: LxRange[<<2, 7>>...<<4, 4>>],
  detail: nil,
  children: []
}
```

Having new lines in the symbol name breaks telescope picker. 

One can argue that this is a problem with the telescope picker and not `lexical`. However, I have looked into how other LSPs return the symbols and if they contain new lines in them. 

For comparison, `elixir-ls` returns the following for the same document:

```
{
  children = {},
  detail = "def",
  kind = 12,
  name = "my_fn/1",
  range = {
    ["end"] = {
      character = 5,
      line = 4
    },
    start = {
      character = 2,
      line = 1
    }
  },
  selectionRange = {
    ["end"] = {
      character = 3,
      line = 3
    },
    start = {
      character = 6,
      line = 1
    }
  }
} 
```

Note that they only include the name and arity of the function in the `name` property.

I have looked at `rust-analyzer` which returns only the name of the function in the `name` property. It adds the argument list into the `detail` field without the new line characters.

I have checked the LSP protocol documentation but couldn't find anything about having new lines in the name property other than saying it should not be an empty string.

I suggest that new lines should not be returned in the `name` field to be consistent with the other LSP implementations.

**Comment by scohen** [comment]
@idursun are you using neovim nightly, or a released version? 

I'm a little wary to change this. Elixir isn't rust, and a function's pattern matching can make it impossible to read on a single line. If the spec says that there shouldn't be newlines then we'll definitely change, but doing what another LS does will lead to chaos in the ecosystem. We all, servers _and_ clients need to follow the spec.

**Comment by idursun** [comment]
I am using a released version of Neovim.

Just to be clear,  Neovim's internal symbol lister (`:lua vim.lsp.buf.document_symbols`) is listing symbols just fine by omitting new line characters.

However, I use the Telescope plugin (as most people do) to display everything and I noticed that it broke when I switched from `elixir-ls` to `lexical`. 

The root cause was that Telescope could not handle new lines in the document symbol list. 

Out of curiosity, I have checked other LSP implementations and noticed that none of them return new lines in the document symbols even though the signature spans multiple lines. That's why I mentioned Rust, Zig etc.

The spec doesn't mention anything about new lines, but they do note that this list will be displayed to the user so it shouldn't contain pure white space. 

I am torn. This sounds like a telescope problem more than it is a lexical problem, however, it would be nice if lexical worked out of the box with one of the most popular plugins in the NeoVim ecosystem.
